.github/workflows/legacy.yml

```
@@ -0,0 +1,32 @@
+# This workflow validates that (new) variables are not referenced as deprecated legacy variables
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Guard against legacy variables
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ '**' ]
+
+jobs:
+  legacy-validation:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
```

There's now version 4 of checkout
```suggestion
    - uses: actions/checkout@v4
```